roles/ceph-common/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+---
+- include: check_system_and_vars.yml
+
+- name: install dependencies
+  apt: pkg={{ item }}
+       state=present
+       update_cache=yes
+       cache_valid_time=3600
+  with_items: ceph_common.debian_pkgs
+
+- name: install ceph
+  apt: pkg={{ item }}
+       state=present
+  with_items: ceph_common.ceph_pkgs
+
+- name: configure rbd clients directories
+  file: path={{ item }}
+        state=directory
+        owner=libvirt-qemu
+        group=kvm
+        mode=0755
+  with_items: ceph_common.rbd_client_paths
+  when: ceph_common.rbd_client_directories
+
+- name: check for a ceph socket
+  shell: "stat /var/run/ceph/*.asok > /dev/null 2>&1"
+  changed_when: false
+  failed_when: false
+  register: socket
+
+# we're storing the fsid in a file on the first monitor node
+# truncate is to get rid of the newline
+- name: generate fsid file if we don't have one
+  shell: uuidgen > /etc/ceph/fsid && truncate -s -1 /etc/ceph/fsid
+         creates=/etc/ceph/fsid
+  run_once: true
+  delegate_to:  "{{ groups['ceph_monitors'][0] }}"
+
+- name: fetch contents of fsid file
+  slurp: path=/etc/ceph/fsid
+  run_once: true
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  register: fsid_file
+
+# for redundancy, and if the hosts in mons get reordered
+- name: copy fsid to all hosts
+  copy:
+    dest: "{{ fsid_file['source'] }}"
+    content: "{{ fsid_file['content'] | b64decode }}"
+
+- name: detect and register public and cluster network
+  shell: ip addr | grep {{ ceph_common.monitor_interface }} | grep inet | awk '{print $2}'
+  run_once: true
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  register: public_and_cluster_network
+
+- name: generate ceph configuration file
+  template: src=ceph.conf.j2
+            dest=/etc/ceph/ceph.conf
+            owner=root
+            group=root
+            mode=0644
+  notify:
+    - restart ceph mons
+    - restart ceph osds
+
```

Comment:
There is no task that enables these services at boot, nor is there a task to ensure that they are started. Using our other roles as example, we have a handler flush for restarts just before a final task in the role that sets the services up to be enabled and started. Since we flush the handlers, the service would have been freshly restarted, so the state=started will be a no-op, and saves an immediate restart after initial start.